db_objects/roster_view.class.php

```diff
@@ -422,8 +422,8 @@ public function printCSV($start_date=NULL, $end_date=NULL, $return=FALSE)
 
 		$to_print = Array();
 		foreach ($services as $id => $service_details) {
-			$to_print[$service_details['date']]['service'][$service_details['congregationid']] = $service_details;
```

Comment:
There's a problem here.  A roster view may include details of several services on the same date (for different congregations). The array needs to be indexed by congregationid or they will overwrite each other.

Reply:
Trying to wrap my head around this.
I've experimented with this on my system and it works with two congregations with services on the same days.
I think it works because getCongregations looks at the members and all the members in the service belong to the same congregation..

Reply:
Ah yes, I think I misread  this line - sorry.  It's already indexed by congregationid.

Reply:
I've changed the order of the indexing and made service details available from all included congregations now.
The code in this PR was pretty stale so I have created a replacement PR #932 after rebasing my fork against your latest version.